components/generic/RightsStatement.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <span data-qa="rights statement">
+    <span
+      v-for="icon in rights.iconClass"
+      :key="icon"
+      :class="icon"
+      class="license"
+    />
+    {{ rights.name }}
+  </span>
+</template>
+
+<script>
+  export default {
+    props: {
+      rightsStatementUrl: {
+        type: String,
+        default: ''
+      }
+    },
+    computed: {
+      rights: function () {
+        const val = this.rightsStatementUrl;
+        let rights = [];
+
+        switch (true) {
+        case new RegExp('https?://creativecommons.org/publicdomain/mark').test(val):
+          rights.name = 'Public Domain';
+          rights.iconClass = ['icon-license-pd'];
+          break;
+        case new RegExp('https?://creativecommons.org/(licenses/)?publicdomain/zero').test(val):
+          rights.name = 'CC0';
+          rights.iconClass = ['icon-license-zero'];
+          break;
+        case new RegExp('https?://creativecommons.org/licenses/by/').test(val):
+          rights.name = 'CC BY';
+          rights.iconClass = ['icon-license-cc'];
+          break;
+        case new RegExp('https?://creativecommons.org/licenses/by-sa').test(val):
+          rights.name = 'CC BY-SA';
+          rights.iconClass = ['icon-license-cc', 'icon-license-by'];
+          break;
+        case new RegExp('https?://creativecommons.org/licenses/by-nc/').test(val):
+          rights.name = 'CC BY-NC';
+          rights.iconClass = ['icon-license-cc', 'icon-license-by', 'icon-license-nc'];
+          break;
+        case new RegExp('https?://creativecommons.org/licenses/by-nc-sa').test(val):
+          rights.name = 'CC BY-NC-ND';
+          rights.iconClass = ['icon-license-cc', 'icon-license-by', 'icon-license-nc', 'icon-license-nd'];
+          break;
+        case new RegExp('https?://creativecommons.org/licenses/by-nc-nd').test(val):
+          rights.name = 'CC BY-NC-SA';
+          rights.iconClass = ['icon-license-cc', 'icon-license-by', 'icon-license-nc', 'icon-license-sa'];
+          break;
+        case new RegExp('https?://creativecommons.org/licenses/by-nd').test(val):
+          rights.name = 'CC BY-ND';
+          rights.iconClass = ['icon-license-cc', 'icon-license-by', 'icon-license-nd'];
+          break;
+        case new RegExp('https?://www.europeana.eu/rights/out-of-copyright-non-commercial').test(val):
+          rights.name = 'No Copyright - Non-Commercial Use Only';
+          rights.iconClass = ['icon-license-rs-no'];
+          break;
+        case new RegExp('https?://rightsstatements.org/vocab/InC-EDU/*').test(val):
+          rights.name = 'In Copyright - Educational Use Permitted';
+          rights.iconClass = ['icon-license-rs-yes'];
+          break;
+        case new RegExp('https?://rightsstatements.org/vocab/NoC-NC/*').test(val):
+          rights.name = 'No Copyright - Non-Commercial Use Only';
```

Comment:
Will these strings be localized (maybe another ticket for that I guess)?

Reply:
Yes, one for later.